### PR TITLE
Read dl2 in hdf5eventsource

### PR DESCRIPTION
**Pull request by maxnoe**
Needed for making `ctapipe-process` applying ml models work on dl2 train files.

### CODE REVIEW

**Comment by kosack** [comment]
Might want to wait for #1899 before merging this (impact parameters in DL2).

**Comment by maxnoe** [comment]
@kosack Done and rebased

**Review by nbiederbeck — APPROVED** [review]

File: ctapipe/io/hdf5eventsource.py [hunk 1]
```diff
@@ -338,7 +350,49 @@ def _generate_events(self):
...
+
+        dl2_tel_readers = {}
+        if DL2_TELESCOPE_GROUP in self.file_.root:
+            dl2_group = self.file_.root[DL2_TELESCOPE_GROUP]
+
+            for kind, group in dl2_group._v_children.items():
+                try:
+                    container = DL2_CONTAINERS[kind]
+                except KeyError:
+                    self.log.warning("Unknown DL2 stereo group %s", kind)
```
Comment (nbiederbeck) [inline ctapipe/io/hdf5eventsource.py:381]: unknown telescope group

**Review by nbiederbeck — APPROVED** [review]

File: ctapipe/io/hdf5eventsource.py [hunk 2]
```diff
@@ -495,6 +547,17 @@ def _generate_events(self):
                             intensity_statistics=simulated_params[4],
                         )
 
+                for kind, algorithms in dl2_tel_readers.items():
+                    c = getattr(data.dl2.tel[tel_id], kind)
+                    for algorithm, readers in algorithms.items():
+                        print(kind, algorithm, tel_id, key)
```
Comment (nbiederbeck) [inline ctapipe/io/hdf5eventsource.py:553]: remove this line

Comment (nbiederbeck) [inline ctapipe/io/hdf5eventsource.py:553]: remove this print function

**Review by nbiederbeck — CHANGES_REQUESTED** [review]

**Comment by maxnoe** [comment]
@nbiederbeck Fixed. This will (if not directly, then at least semantically) conflict with #1949.

I would propose to merge this PR first to get the basic functionality in and then I can rebase the other one.

**Review by nbiederbeck — APPROVED** [review]

**Review by LukasNickel — APPROVED** [review]

**Pull request merged**